capitalone/rubicon-ml
# add tagging to artifacts

closes: #244 

---

## What
  * add tagging to artifacts
    * applied the `TagMixin` to the artifact domain object
    * updated client and repo layer to reflect `TagMixin` addition
      * updated `_get_tag_metadata_root` to handle all rubicon objects
  * updated tests

## How to Test
  * run through the quick look examples and validate they work as expected
  * try logging tags to artifacts
 
```python
from rubicon_ml import Rubicon

rubicon = Rubicon(persistence="filesystem", root_dir=".")

# test tagging project artifacts
project = rubicon.get_or_create_project(name="test artifacts")
project.log_artifact(data_bytes=b"bytes", name="name", tags=["tags_a"])

# test tagging experiment artifacts
experiment = project.log_experiment()
experiment.log_artifact(data_bytes=b"bytes", name="name", tags=["tags_b"])
```
